### PR TITLE
dev: clean up dev docker-compose

### PR DESCRIPTION
**Pull request by ghost**
- Move docker-compose overrides files to docker folder
- Add scripts/osrd-compose.sh to start docker-compose with host overrides

### CODE REVIEW

File: scripts/osrd-compose.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+# This script should be compatible with MSYS, the compatibility layer used by
+# Git for Windows. Absolute paths which should not be converted to windows paths
+# have to start with //, see https://github.com/git-for-windows/git/issues/1387
+# On windows, docker cp does not like leading double / on the container path.
+# As a workaround, use relative paths: container:tmp/foo instead of container://tmp/foo
```
Comment (flomonster) [inline scripts/osrd-compose.sh]:
```suggestion
```

Maybe we should warn users that this script will only work on Linux.

Comment (ghost) [reply scripts/osrd-compose.sh]: The command `docker compose` isn't supported on Windows systems ? (I've not used a Windows for anything docker related for a while so I do not know how it behaves at all).

Comment (flomonster) [reply scripts/osrd-compose.sh]: Docker compose is supported, but `network_mode: host` isn't.

File: docker/init_db.sql
Comment (flomonster) [inline docker/init_db.sql:1]: We need to adapt `.github/workflows/editoast.yml`.

File: docker/docker-compose.host.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  core:
+    network_mode: host
+    environment:
+      CORE_PORT: 8080
+      MIDDLEWARE_BASE_URL: "http://localhost:8090"
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost:8080/health"]
+
+  editoast:
```
Comment (flomonster) [inline docker/docker-compose.host.yml:20]: We could override `ports` to avoid warning running this docker-compose file.

**Review by flomonster — CHANGES_REQUESTED** [review]
Great PR!

I have had an issue using `./osrd-compose.sh`. Core is listening on port `80` and not `8080`.

We can either:
- Adapt `docker-compose.yml` so core listens on `8080` and we forward the ports as `8080:8080`.
- Adapt `docker-compose.host.yml` by overriding the command executed by core.

File: docker/docker-compose.host.yml (hunk 2 above)
Comment (ghost) [reply docker/docker-compose.host.yml:20]: Overriding `ports` with something like `ports: []` does not prevent the warning. Do you have something else in mind ?

**Comment by ghost** [comment]
I've introduced a `CORE_PORT` variable that is used in the CMD. Since the `ports` elements are simply discarded by the `network_mode: host` it is not a big deal if the `ports` variable is still defined to the value `[ "8080:80" ]`.

File: docker/docker-compose.host.yml (hunk 2 above)
Comment (flomonster) [reply docker/docker-compose.host.yml:20]: Never mind, I thought it would work.

**Review by flomonster — APPROVED** [review]
LGTM. Reviewed + Tested

**Review by multun — APPROVED** [review]

**Pull request merged**